.github/workflows/run-tests-and-checks.yml

```diff
@@ -0,0 +1,38 @@
+name: Run tests and checks
+
+on:
+  push:
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['3.1', '3.0', '2.7']
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
```

Comment:
I believe this flag runs `bundle install` automatically according to this statement `bundler-cache: true # runs 'bundle install' and caches installed gems automatically` from https://github.com/ruby/setup-ruby#matrix-of-ruby-versions

so we can exclude teh install deps step.

Reply:
Good point, I hadn't seen that 💯 Now when I rebase this PR on top of the other branch I'll make that change too 🙇‍♀️

Reply:
done